src/k8s-engine/wsl.ts

```diff
@@ -307,6 +308,39 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
         this.emit('service-changed', services);
       });
       this.activeVersion = desiredVersion;
+
+      // Temporary workaround: ensure root is mounted as shared -- this will be done later
+      // Right now the builder pod needs to be restarted after the remount
+      try {
+        await childProcess.exec('wsl --user root -d k3s mount --make-shared /');
+        console.log('Waiting for ensuring root is shared');
+        await util.promisify(setTimeout)(60_000);
+        await childProcess.execFile(resources.executable('kim'), ['builder', 'install', '--force', '--no-wait']);
+        const startTime = Date.now();
+        const maxWaitTime = 120_000;
+        const waitTime = 3_000;
+
+        while (true) {
+          const currentTime = Date.now();
+
+          if ((currentTime - startTime) > maxWaitTime) {
+            console.log(`Waited more than ${ maxWaitTime / 1000 } secs, it might start up later`);
+            break;
+          }
+          // Find a working pod
+          const pod = await this.client.getActivePod('kube-image', 'builder');
+
+          console.log(`QQQ: got pod ${ pod }, status: ${ pod?.status?.phase }`);
+          if (pod?.status?.phase === 'Running') {
+            break;
+          }
+          await util.promisify(setTimeout)(waitTime);
+        }
+      } catch (e) {
+        console.log(`Failed to restart the kim builder: ${ e.message }.`);
+        console.log('The images page will probably be empty');
+      }
+      await util.promisify(setTimeout)(30 * 1000);
```

Comment:
Not sure why we need the extra wait here?

Reply:
Don't think we do. Let me test that.